crates/anvil/tests/it/otterscan.rs

```diff
@@ -88,170 +75,88 @@ async fn can_call_ots_get_internal_operations_contract_transfer() {
-        }
+            from: address!("4e59b44847b379578588920cA78FbF26c0B4956C"),
+            to: address!("347bcdad821abc09b8c275881b368de36476b62c"),
+            value: U256::from(0),
+        }],
     );
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn can_call_ots_get_internal_operations_contract_selfdestruct() {
-    let prj = TempProject::dapptools().unwrap();
-    prj.add_source(
-        "Contract",
-        r"
-pragma solidity 0.8.13;
-contract Contract {
-    address payable private owner;
-    constructor() public {
-        owner = payable(msg.sender);
-    }
-    function goodbye() public {
-        selfdestruct(owner);
-    }
-}
-",
-    )
-    .unwrap();
-
+async fn ots_get_internal_operations_contract_selfdestruct() {
     sol!(
-        #[sol(rpc)]
+        #[sol(rpc, bytecode = "60808060405234601357607c908160188239f35b5f80fdfe6004361015600b575f80fd5b5f3560e01c6375fc8e3c14601d575f80fd5b346042575f36600319011260425773dcdd539da22bffaa499dbea4d37d086dde196e75ff5b5f80fdfea26469706673582212201778e88721238015f480b3a901848b460799e91836caf557a8dcfdb2a1a2469e64736f6c63430008190033")]
         contract Contract {
-            function goodbye() external;
+            function goodbye() public {
+                selfdestruct(payable(0xDcDD539DA22bfFAa499dBEa4d37d086Dde196E75));
+            }
         }
     );
 
-    let mut compiled = prj.compile().unwrap();
-    assert!(!compiled.has_compiler_errors());
-    let contract = compiled.remove_first("Contract").unwrap();
-    let bytecode = contract.into_bytecode_bytes().unwrap();
-
     let (api, handle) = spawn(NodeConfig::test()).await;
-    let wallet = handle.dev_wallets().next().unwrap();
-    let signer: EthereumSigner = wallet.clone().into();
-    let sender = wallet.address();
-
-    let provider = ws_provider_with_signer(&handle.ws_endpoint(), signer);
-
-    // deploy successfully
-    provider
-        .send_transaction(WithOtherFields::new(TransactionRequest {
-            from: Some(sender),
-            to: Some(TxKind::Create),
-            input: bytecode.into(),
-            ..Default::default()
-        }))
-        .await
-        .unwrap()
-        .get_receipt()
-        .await
-        .unwrap();
-    let contract_address = sender.create(0);
-    let contract = Contract::new(contract_address, &provider);
+    let provider = handle.http_provider();
+
+    let contract = Contract::deploy(&provider).await.unwrap();
 
     let receipt = contract.goodbye().send().await.unwrap().get_receipt().await.unwrap();
 
     let res = api.ots_get_internal_operations(receipt.transaction_hash).await.unwrap();
-
-    assert_eq!(res.len(), 1);
     assert_eq!(
-        res[0],
-        OtsInternalOperation {
+        res,
+        [OtsInternalOperation {
             r#type: OtsInternalOperationType::SelfDestruct,
             from: *contract.address(),
-            to: Default::default(),
-            value: U256::from(0)
-        }
+            to: Address::ZERO,
```

Comment:
This was already zero but I'm not sure if this is correct, as selfdestruct technically has a `to` address. cc @sealer3

Reply:
You're right - for selfdestruct operations, the `type`, `from`, `to`, and `value` should all be given an `ots_getInternalOperations` response, using the smart contract being self-destructed as the `from` address and the target of the selfdestruct as the `to` address.